### ISSUE TITLE
Add option to Style/SymbolProc to ignore methods with parameters

### ISSUE THREAD

**Issue by stephannv** [rubocop/rubocop#9525]
## Is your feature request related to a problem? Please describe.

No.

## Describe the solution you'd like

My application has some code like this:

```ruby
on_scheduled_jobs(worker_class_name, some_arg) do |job|
  job.delete
end

after_transition(after_commit: true, to: :deleted) do |project|
  project.clear_indexes
end
```

Rubocop detects offense and suggests change my code to:
```ruby
on_scheduled_jobs(worker_class_name, first_arg_c, &:delete)

after_transition(after_commit: true, to: :deleted, &:clear_indexes)
```

But in some scenarios, using symbol as proc isn't the best option for code style because don't follow other block definitions style. I know I have some alternatives like disable `Style/SymbolProc`, add methods to IgnoredMethods option or use `# rubocop:disable Style/SymbolProc` for some cases, but I think a option to ignore methods with parameters would be nice.

## Describe alternatives you've considered
Add option (maybe a EnforcedStyle??) to ignore `Style/SymbolProc` for methods with parameters, or option to only enable `Style/SymbolProc` for methods without parameters.

For example:
```yaml
# rubocop.yml
Style/SymbolProc:
  IgnoreMethodsWithParameters: true
```
```ruby
# code.rb
users.each { |u| u.send_email } # Detects offense

after_transition(to: :paid) do |payment| # Doesn`t detect offense
   payment.send_notification
end
  
```
